src/workflows/workflows.ts

```diff
@@ -78,14 +82,25 @@ export class Workflows {
     this.withdrawalsApi = new WithdrawalsApi(config.apiConfiguration);
   }
 
-  public registerOffchain(walletConnection: WalletConnection) {
+  private async validateChain(signer: Signer) {
+    if (!this.isChainValid(await signer.getChainId()))
```

Comment:
to have await as a param, bad idea. can you please extract it to variable

Reply:
no probs. Done.